### PR TITLE
Fix #17 - Use core pagination component

### LINKED ISSUE #17: consume PF4 pagination from core
need to update Pagination imports and a small refactor
in order to use the new PF4 pagination from core.
https://github.com/theforeman/foreman/pull/8990

### PR DESCRIPTION
**Pull request by ofedoren**
@Ron-Lavi, is this correct? :)

### CODE REVIEW

**Comment by ofedoren** [comment]
rubocop issues are fixed here: https://github.com/theforeman/foreman_host_reports/pull/19, but I can do it as part of this PR as well.

File: lib/foreman_host_reports/engine.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@ class Engine < ::Rails::Engine
 
     initializer 'foreman_host_reports.register_plugin', :before => :finisher_hook do |_app|
       Foreman::Plugin.register :foreman_host_reports do
-        requires_foreman '>= 3.0.0'
+        requires_foreman '>= 3.2.0'
```
Comment (ofedoren) [inline lib/foreman_host_reports/engine.rb:21]: @lzap, I guess you need to know about that :/ @Ron-Lavi, I guess we're not going to backport this to one of the previous versions of Foreman?

File: webpack/src/Router/HostReports/IndexPage/Components/HostReportsTable/HostReportsTable.js [hunk 2]
```diff
@@ -55,11 +54,11 @@ const HostReportsTable = ({
...
       {body}
       <Pagination
-        viewType="list"
         itemCount={itemCount}
-        pagination={pagination}
-        onChange={fetchAndPush}
-        dropdownButtonId="host-reports-page-pagination-dropdown"
+        /* eslint-disable-next-line camelcase */
+        onChange={({ page, per_page }) =>
+          fetchAndPush({ page, perPage: per_page })
```
Comment (Ron-Lavi) [inline webpack/src/Router/HostReports/IndexPage/Components/HostReportsTable/HostReportsTable.js]: I believe this should be `onChange={fetchAndPush}`
and in `fetchAndPush` we should use only `per_page`
as in https://github.com/theforeman/foreman_host_reports/blob/develop/webpack/src/Router/HostReports/IndexPage/IndexPageActions.js#L36 we are sending `per_page` to the server.

**Review by Ron-Lavi — COMMENTED** [review]
Great work @ofedoren! thanks for doing this
tested and it works as expected, left one comment

File: lib/foreman_host_reports/engine.rb (hunk 1 above)
Comment (Ron-Lavi) [reply lib/foreman_host_reports/engine.rb:21]: I guess we are not going to backport it in core,
but I guess this plugin will be official from 3.2? cc @lzap

File: webpack/src/Router/HostReports/IndexPage/Components/HostReportsTable/HostReportsTable.js (hunk 2 above)
Comment (ofedoren) [reply webpack/src/Router/HostReports/IndexPage/Components/HostReportsTable/HostReportsTable.js]: Alright then, I'll update that here and in webhooks as well :)

**Comment by ofedoren** [comment]
@Ron-Lavi, updated.

**Review by Ron-Lavi — APPROVED** [review]
works as expected, thanks @ofedoren 👍🏼

**Pull request merged**